### PR TITLE
Add null default serializer in order to support ignore annotations

### PR DESCRIPTION
**Pull request by atrianac**
Add null default serializer in order to support ignore annotations

### CODE REVIEW

**Comment by ruslansennov** [comment]
Hi Alejandro
Could you add tests to show clarify the problem?

File: src/main/java/javaslang/jackson/datatype/serialize/OptionSerializer.java [hunk 1]
```diff
@@ -29,6 +31,15 @@
...
+        if(value.isEmpty()) {
+            provider.defaultSerializeNull(gen);
+        } else {
+            gen.writeObject(toJavaObj(value));
+        }
+    }
+
+    @Override
     Object toJavaObj(Option<?> value) throws IOException {
         return value.isEmpty() ? null : value.get();
```
Comment (ruslansennov) [inline src/main/java/javaslang/jackson/datatype/serialize/OptionSerializer.java]: I believe now this should be <code>return value.get();</code>

**Comment by atrianac** [comment]
@ruslansennov I updated the pull request. The problem is that OptionalSerializer ignores @JsonInclude(NON_EMPTY) or @JsonInclude(NON_NULL).

**Comment by ruslansennov** [comment]
@atrianac thank you!
I believe we can move this check to <code>ValueSerializer</code>
```java
abstract class ValueSerializer<T> extends StdSerializer<T> {

    // ...

    @Override
    public void serialize(T value, JsonGenerator gen, SerializerProvider provider) throws IOException {
        if(isEmpty(provider, value)) {
            provider.defaultSerializeNull(gen);
        } else {
            gen.writeObject(toJavaObj(value));
        }
    }
}
```
Because <code>JsonSerializer::isEmpty(SerializerProvider provider, Option&lt;?&gt; value)</code> overridden in <code>OptionSerializer</code>
Am I right?

Also <code>OptionSerializer::toJavaObj(Option&lt;?&gt; value)</code> can be simplified:
```java
    @Override
    Object toJavaObj(Option<?> value) throws IOException {
        return value.get();
    }

```

**Comment by atrianac** [comment]
@ruslansennov I did the changes. Thanks

**Comment by ruslansennov** [comment]
Fixes #51

**Pull request merged**